mongodb/connector/connection/connection.go

```diff
@@ -0,0 +1,250 @@
+						pkey, _ = x509.DecryptPEMBlock(v, []byte(KeyPass))
+						pkey = pem.EncodeToMemory(&pem.Block{
+							Type:  v.Type,
+							Bytes: pkey,
+						})
+					} else {
+						pkey = pem.EncodeToMemory(v)
+					}
+				}
+				cert, err = tls.X509KeyPair([]byte(certPEMBlock), pkey)
+			} else {
+				cert, err = tls.X509KeyPair([]byte(certPEMBlock), []byte(keyPEMBlock))
+			}
+			if err != nil {
+				logmongoconn.Errorf("Error while creating client certs for establishing 2 way SSL", err)
+				return nil, err
+			}
+			tlsConfig = &tls.Config{
+				RootCAs: roots,
+				//	ClientAuth: tls.RequireAndVerifyClientCert,
+				//	ClientCAs:  clients,
+				Certificates: []tls.Certificate{cert},
+			}
+			// x-509 implementation
+			x509 := sharedConn.config.X509
+			if x509 {
+				opts.SetAuth(options.Credential{
+					AuthMechanism: "MONGODB-X509",
+					AuthSource:    "$external",
+				})
+			}
+		}
+		opts.SetTLSConfig(tlsConfig)
+
+	}
+	client, err := mongo.NewClient(opts.ApplyURI(url))
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	err = client.Connect(ctx)
+	if err != nil {
+		logmongoconn.Errorf("===connect error==", err)
+		return nil, err
+	}
+	err = client.Ping(ctx, nil)
+	if err != nil {
+		logmongoconn.Errorf("===ping error===", err)
+	} else {
+		logmongoconn.Debugf("===Ping success===")
+		sharedConn.mclient = client
+		logmongoconn.Debugf("Returning new mongodb connection")
+	}
+	return sharedConn, nil
+}
+
+// MongodbSharedConfigManager Structure
+type MongodbSharedConfigManager struct {
+	config  *Settings
+	name    string
+	mclient *mongo.Client
+}
+
+// Type of SharedConfigManager
+func (k *MongodbSharedConfigManager) Type() string {
+	return "mongodb"
+}
+
+// GetConnection ss
+func (k *MongodbSharedConfigManager) GetConnection() interface{} {
+	return k
+}
+
+// GetClient type
+func (k *MongodbSharedConfigManager) GetClient() *mongo.Client {
+	return k.mclient
+}
+
+// GetClientConfiguration of mongo connection
+func (k *MongodbSharedConfigManager) GetClientConfiguration() *Settings {
+	return k.config
+}
+
+// ReleaseConnection ss
+func (k *MongodbSharedConfigManager) ReleaseConnection(connection interface{}) {
+
+}
+
+// Start connection manager
+func (k *MongodbSharedConfigManager) Start() error {
+	return nil
+}
+
+// Stop connection manager
+func (k *MongodbSharedConfigManager) Stop() error {
+	logmongoconn.Debug("Cleaning up client connection cache")
+	return nil
+}
+
+// GetSharedConfiguration function to return MongoDB connection manager
+func GetSharedConfiguration(conn interface{}) (connection.Manager, error) {
+	var cManager connection.Manager
+	var err error
+	cManager, err = coerce.ToConnection(conn)
+	if err != nil {
+		return nil, err
+	}
+	return cManager, nil
+}
+
+func getmongodbClientConfig(settings map[string]interface{}) (*Settings, error) {
+	connectionConfig := &Settings{}
+
+	s := &Settings{}
```

Comment:
You don't need this.

Reply:
yes, when I reviewed, I knew it had to go. I thought I did it..Will be removed in the next git push